### PR TITLE
✨🥗 `Marketplace`: `VendorRepresentative` manages `Products`

### PR DESCRIPTION
**Pull request by zspencer**
- https://github.com/zinc-collective/convene/issues/2044
- https://github.com/zinc-collective/convene/issues/1324

OK this should probably be split into a number of smaller steps, each with reasonable test coverage; but I wanted to get something working so we can turn the keys over to the folks at Oaklandia.

This makes it so:

- A `Neighborhood` `Operator` or `Space` `Member` can add a `VendorRepresentative` to a `Marketplace`
- The `VendorRepresentative` can add, remove, and update `Products`

YOLO YOLO YOLO

### CODE REVIEW

File: app/furniture/marketplace/vendor_representative.rb
Comment (rosschapman) [inline app/furniture/marketplace/vendor_representative.rb:7]: Record creation is failing because this is not optional. Observed this while playing with the branch locally.

**Review by rosschapman — APPROVED** [review]

**Comment by rosschapman** [comment]
@zspencer just want to confirm that a vendor rep must be _claimed_ before they actually have any write access to products etc... right? I think that workflow makes sense. Also a Person is generally created through an invitation to membership, right?

File: app/furniture/marketplace/vendor_representative.rb
Comment (zspencer) [reply app/furniture/marketplace/vendor_representative.rb:7]: Weird; I was able to create a `VendorRepresentative` through the UI...

I'll add a `spec/marketplace/system/vendor_representatives_system_spec.rb` and a bit more tests tonight before merging.

Comment (anaulin) [reply app/furniture/marketplace/vendor_representative.rb:7]: Certainly as written now this class requires to be associated to a `person` to be valid.

**Review by anaulin — APPROVED** [review]
I guess this is fine if it gets the job done.

I'm a bit confused about the intended workflow here. So the marketplace owner adds an owner representative, then talks to them to set up an account, then goes back to the marketplace and "claims" them, is that the intended flow?

Why not do something simpler, where a VendorRepresentative is just a bridge between a marketplace and a person, and the Marketplace owner can add it by selecting from a list of People who are members of the Space? (and if the person they want to add is not a member, they can send an invitation, making that person a member, right?)

File: app/furniture/marketplace/vendor_representative.rb
Comment (zspencer) [reply app/furniture/marketplace/vendor_representative.rb:7]: yea, adding the spec caused it to fail until I put this in... Now I'm wondering why I could do it last night...

**Comment by zspencer** [comment]
> Why not do something simpler, where a VendorRepresentative is just a bridge between a marketplace and a person, and the Marketplace owner can add it by selecting from a list of People who are members of the Space? (and if the person they want to add is not a member, they can send an invitation, making that person a member, right?)

Right now, being a `Member` grants wide permissions across the `Space`; we want to keep the permissions to *only* the `Marketplace`. So we can't add a `Person` from their `Membership`, since that would escalate their permissions too highly.

> I'm a bit confused about the intended workflow here. So the marketplace owner adds an owner representative, then talks to them to set up an account, then goes back to the marketplace and "claims" them, is that the intended flow?

I wanted to figure out a way to link it directly and not have to do this awkward handshake; but I couldn't come up with a way to do it without pushing the `Vendor` into `Marketplace` land. It's not ideal :(.

**Comment by zspencer** [comment]
> @zspencer just want to confirm that a vendor rep must be _claimed_ before they actually have any write access to products etc... right? I think that workflow makes sense. Also a Person is generally created through an invitation to membership, right?

A `Person` is created whenever someone signs in to a `Space` in the `Neighborhood`. We probably want to move so that "`Person` registers in `Neighborhood`" is it's own distinct workflow tho, since I now understand the folly of conflating sign up and sign in.

**Pull request merged**
